### DISCUSSION TITLE
components.css overwrites my custom theme

### DISCUSSION THREAD

**Discussion by iamolivinius** [primefaces/community#1055] [PrimeFaces] [answered]
Hi all!

Is there a way how i can change the loading order of css resources? 

I'm writing a custom theme but some of my changes get overwritten by components.css. Is there another way around this issue but using !important all over the place?

Keep up the good work :)

**Comment by noahbazer** [comment, marked as answer]
Hey! Instead of writing multiple styles and changing their load order, you should get into the habit of using very specific class selectors (or ID selectors) whenever possible. Not only will this cut down on your code footprint, but will reduce load times as well (even if barely). 
Here's an example of that specificity in action:

For this HTML: 
```
<div class="container">
  <div class="element">
    <p>This is some text</p>
  </div>
</div>
```

Instead of doing this:
```
div {
  color: red;
  font-size: 1.2em;
}

div div {
  color: blue;
}

div p {
  color: green;
}
```

Do this:
```
.container {
  color: red;
  font-size: 1.2em;
}

.element {
  color: blue;
}

.container .element p {
  color: green;
}
```

Not only does this prevent styles from overriding each other, but your code will be easier to read and maintain. Hope this helps!

**Reply by jepsar** [reply]
@iamolivinius For more info, please read https://stackoverflow.com/questions/8768317/how-do-i-override-default-primefaces-css-with-custom-styles

**Comment by noahbazer** [comment]
If you want help with your specific code, post your code here and we can help you further :)

**Comment by melloware** [comment]
Agreed with @noahbazer you need to make your style overrides as "specific" or "more specific" CSS.  In CSS its all about specificity levels!
